R/PseudoBulk.R

```diff
@@ -526,14 +526,23 @@ RunFilteredContrasts <- function(seuratObj, filteredContrastsFile = NULL, filter
   }
   
   results <- future.apply::future_lapply(split(contrasts, 1:nrow(contrasts)), future.seed = GetSeed(), FUN = function(x){
-    #initialize two seurat objects, one to be subset according to the positive contrasts, and one to be subset according to the negative contrast. These will be merged downstream.
+    #initialize two Seurat objects, one to be subset according to the positive contrasts, and one to be subset according to the negative contrast. These will be merged downstream.
     seuratObj.positive.contrast <- seuratObj
     seuratObj.negative.contrast <- seuratObj
     positive_contrast <- NULL
     negative_contrast <- NULL
     #construct the positive and negative contrast by accessing each value within the filtered contrast data frame (row-wise) by adding the appropriate prefix to the metadata field name (i.e. the "contrast column")
     print(paste0("Contrast columns: ", attr(design,"contrast_columns")))
     for (contrast_column in attr(design, "contrast_columns")){
+      #check if the contrast column in the parent Seurat object needs sanitizing before populating seuratObj.positive.contrast and seuratObj.negative.contrast downstream.
+      if (!all(seuratObj@meta.data[,contrast_column] == gsub("_", ".", make.names(seuratObj@meta.data[,contrast_column])))){
+        print("Converting metadata columns to a make.names() format. Hyphens, spaces, underscores, and other non-alphanumeric characters will be converted to periods. Factor levels will be retained.")
+        if (is.factor(seuratObj@meta.data[,contrast_column])) {
```

Comment:
Given you are replicating a not entirely trivial bit of code here and also lines 541, 563 and 569, could you make an internal function, like ".RemoveSpecialCharacters(seuratObj, fieldName)", which does this work using one central piece of code?

Reply:
yeah, that's better. will do.